src/pybind/mgr/devicehealth/module.py

```diff
@@ -0,0 +1,257 @@
+
+"""
+Device health monitoring
+"""
+
+import errno
+import json
+from mgr_module import MgrModule, CommandResult
+import rados
+from threading import Event
+from datetime import datetime, timedelta, date, time
+
+TIME_FORMAT = '%Y%m%d-%H%M%S'
+
+class Module(MgrModule):
+    OPTIONS = [
+        { 'name': 'active' },
+        { 'name': 'scrape_frequency' },
+        { 'name': 'pool_name' },
+        { 'name': 'retention_period' },
+    ]
+    DEFAULTS = {
+        'active': True,
+        'scrape_frequency': str(86400),
+        'retention_period': str(86400*14),
+        'pool_name': 'device_health_metrics',
+    }
+    active = DEFAULTS['active']
```

Comment:
These 4 lines + the run, event, last_scrape_time below would more naturally be instance attributes (set in __init__) rather than class attributes.  It kind of doesn't matter when we only ever have one instance of the class, but it's more readable that way.